Example/Example-iOS/SceneDelegate.swift

```diff
@@ -10,16 +10,74 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 import UIKit
+import Firebase
+import OktaLogger
+import OktaAnalytics
+import AppCenterAnalytics
+
+var scenarioID: ScenarioID = ""
 
 class SceneDelegate: UIResponder, UIWindowSceneDelegate {
 
     var window: UIWindow?
 
+    let appCenterAnalyticsProvider: AnalyticsProviderProtocol = {
+        let logger = OktaLogger()
+        logger.addDestination(
+            OktaLoggerConsoleLogger(
+                identifier: "com.okta.loggerDemo.console",
+                level: OktaLoggerLogLevel.debug,
+                defaultProperties: nil
+            )
+        )
+        let appCenterAnalyticsProvider = AppCenterAnalyticsProvider(name: "AppCenter", logger: logger)
+        appCenterAnalyticsProvider.start(withAppSecret: "App Secret", services: [AppCenterAnalytics.Analytics.self])
+        return appCenterAnalyticsProvider
+    }()
+
     func scene(_ scene: UIScene, willConnectTo session: UISceneSession, options connectionOptions: UIScene.ConnectionOptions) {
         // Use this method to optionally configure and attach the UIWindow `window` to the provided UIWindowScene `scene`.
         // If using a storyboard, the `window` property will automatically be initialized and attached to the scene.
         // This delegate does not imply the connecting scene or session are new (see `application:configurationForConnectingSceneSession` instead).
+        FirebaseApp.configure()
+        OktaAnalytics.addProvider(appCenterAnalyticsProvider)
+        OktaAnalytics.trackEvent("applicationDidFinishLaunchingWithOptions", withProperties: nil)
+        OktaAnalytics.startScenario(ScenarioEvent(name: "Application", properties: [Property(key: "AppDelegate.application.didFinishLaunchingWithOptions", value: "1")])) {
+            scenarioID = $0 ?? ""
+        }
         guard let _ = (scene as? UIWindowScene) else { return }
     }
+    func sceneWillEnterForeground(_ scene: UIScene) {
+        if scenarioID.isEmpty {
+            OktaAnalytics.startScenario(ScenarioEvent(name: "Application", properties: [Property(key: "AppDelegate.application.sceneWillEnterForeground", value: "1")])) {
+                scenarioID = $0 ?? ""
+            }
+        } else {
+            OktaAnalytics.updateScenario(scenarioID, [Property(key: "SceneDelegate.sceneWillEnterForeground", value: "2")])
```

Comment:
Again, all calls to analytics are blocking and time consuming. Analytics calls should be lightweight and all heavy processing should be done in background thread. It is  unacceptable for Analytics framework to block currently executing thread
In this example we are blocking executing thread with two expensive calls:  startScenario and updateScenario.